Amazon.IonHashDotnet/Serializer.cs

```diff
@@ -116,12 +117,17 @@ internal byte[] Digest()
             return this.HashFunction.Digest();
         }
 
-        internal void HandleFieldName(string fieldName)
+        internal void HandleFieldName(IIonHashValue ionValue, bool isInStruct)
         {
-            // the "!= null" condition allows the empty symbol to be written
-            if (fieldName != null && this.depth > 0)
+            if (this.depth > 0 && isInStruct)
             {
-                this.WriteSymbol(fieldName);
+                if (ionValue.CurrentFieldNameSymbol.Text == null &&
+                    ionValue.CurrentFieldNameSymbol.Sid == 10)
```

Comment:
What is SID 10?

Reply:
SID10 is an unresolved SID which is expected to throw: https://github.com/byronlin-bq/ion-hash-dotnet/blob/master/Amazon.IonHashDotnet.Tests/IonHashWriterTest.cs#L125-L126

Discussed with Tyler and he said the GetFieldNameSymbol() api for binary reader is not expected to throw so we need to add the check in IonHash. Will confirm with Peter.

https://github.com/amzn/ion-dotnet/blob/master/Amazon.IonDotnet/Internals/Binary/SystemBinaryReader.cs#L226-L233